app/scripts/motions.directive.js

```diff
@@ -13,18 +13,58 @@
 angular.module('dashboard')
     .directive('dbMotions', [function () {
 
-        var controller = ['$log', 'StorageSrv', 'CONST', '$rootScope', '$scope', function ($log, StorageSrv, CONST, $rootScope, $scope) {
+        var controller = ['$log', 'StorageSrv', 'CONST', '$rootScope', '$scope', 'AhjoMeetingSrv', 'Utils', function ($log, StorageSrv, CONST, $rootScope, $scope, AhjoMeetingSrv, Utils) {
             $log.log("dbMotions: CONTROLLER");
             var self = this;
             self.loading = false;
             self.motions = null;
-            self.errorCode = 0;
+            self.error = null;
             self.isTooltips = $rootScope.isTooltips;
             self.meetingActive = null;
             var selectedMotion = null;
+            var mtgItemSelected = StorageSrv.getKey(CONST.KEY.MEETING_ITEM);
 
             // FUNCTIONS
 
+            function sign(aMotion, aMeetingGuid) {
+                if (!angular.isObject(aMotion) || !angular.isString(aMeetingGuid)) {
+                    $log.error("dbMotions.sign: bad args", arguments);
+                    return;
+                }
+                $log.log("dbMotions: sign", arguments);
+                AhjoMeetingSrv.signMotion(aMotion.motionGuid, aMotion.personGuid, aMeetingGuid).then(function (resp) {
+                    if (angular.isObject(resp)) {
+                        $log.log("dbMotions.sign done", resp);
+                        aMotion.isUserSupported = resp.isSupported;
+
+                        var supporter = {
+                            firstName: null,
+                            insertDateTime: null,
+                            lastName: null,
+                            motionGuid: aMotion.motionGuid,
+                            name: aMotion.personName,
+                            personGuid: aMotion.personGuid
+                        };
+                        if (angular.isArray(aMotion.supporters)) {
+                            aMotion.supporters.push(supporter);
+                        }
+                        else {
+                            aMotion.supporters = [supporter];
+                        }
+                    }
+                    else {
+                        $log.error("dbMotions.sign done ", resp);
+                    }
+                }, function (error) {
+                    $log.error("dbMotions.sign ", error);
+                    self.error = { 'errorCode': error.errorCode, 'errorString': Utils.stringIdForError(error.errorCode) };
```

Comment:
This is OK by me, but just remarking how there's `GENERALERROR` in `CONST` which could be extended by `errorStringId` and/or `errorString` and reused.